### PR TITLE
fix(android/engine): Test fileVersion during package install

### PR DESCRIPTION
**Pull request by darcywong00**
Addresses #6254 for 15.0 beta. Will need to 🍒 pick to stable.14.0 and master to fix the issue.

This updates PackageActivity() to check kmp.json that the minimum keyboard version (system.fileVersion) is supported by the current Keyman version.

### Notes on test keyboards
The test keyboard "the_99.kmp" is a standard keyboard created from template where fileVersion is manually edited to "99.0". 
Keyman Developer currently (as of 15.0.210 beta) doesn't allow `store(&VERSION)` to define version greater than '15.0'.

Also updated full_caps_3620_3621 to use `store(&VERSION) '15.0'`

## User Testing

Setup
1. On an Android device/emulator, load the PR build of Keyman for Android
2. Also upload the [test_99.kmp](https://github.com/keymanapp/keyman/raw/b213c3b8291221746d96013769308430d7716f04/common/test/keyboards/the_99/the_99.kmp) keyboard to the device's "Download" folder.

* **TEST_FILEVERSION_CHECKED**
1. Launch Keyman
4. Dismiss the "Get Started" menu
5. From the "Settings" menu --> "Install Keyboard or Dictionary" --> Install from local file 
6. Browse to the Downloads folder where the test_99.kmp file was uploaded.
7. Install the test_99.kmp keyboard and verify the installation fails
Screenshot:
 
unsupported

8. From the "Settings" menu, install the khmer_angkor keyboard from keyman.com
9. Verify the khmer_angkor keyboard downloads and installs.

### CODE REVIEW

File: common/test/keyboards/full_caps_3620_3621/source/full_caps_3620_3621.kmn [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿c full_caps_3620_3621 generated from template at 2021-12-01 08:45:15
 c with name "full_caps_3620_3621"
-store(&VERSION) '10.0'
+store(&VERSION) '15.0'
```
Comment (ermshiperete) [inline common/test/keyboards/full_caps_3620_3621/source/full_caps_3620_3621.kmn:3]: Why this change? Is this another test case for this PR, or does this keyboard use features introduced in 15.0?

We'll have to adjust this for stable-14.

**Review by ermshiperete — CHANGES_REQUESTED** [review]
Can we check in the sources for `the_99`?

File: common/test/keyboards/full_caps_3620_3621/source/full_caps_3620_3621.kmn (hunk 1 above)
Comment (darcywong00) [reply common/test/keyboards/full_caps_3620_3621/source/full_caps_3620_3621.kmn:3]: The U_xxxx_yyyy is a 15.0 feature which will cause the 14.0 Keyman app to crash trying to render the popup keys.
I presume the Windows build is now failing due to a test. will chase that down.

Comment (darcywong00) [reply common/test/keyboards/full_caps_3620_3621/source/full_caps_3620_3621.kmn:3]: The CI fail is cause all the common/test/keyboards are expected to build so it's wanting the keyboard project...

I didn't check it in cause I had to manually edit the built kmp.json file...

Comment (mcdurdin) [reply common/test/keyboards/full_caps_3620_3621/source/full_caps_3620_3621.kmn:3]: I am about to add a child PR to this branch that fixes the build issue for common/test/keyboards

Comment (mcdurdin) [reply common/test/keyboards/full_caps_3620_3621/source/full_caps_3620_3621.kmn:3]: See #6358

File: android/KMAPro/kMAPro/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -241,4 +241,7 @@
   <!-- Context: KMP Package strings -->
   <string name="invalid_metadata" comment="kmp.json metadata file is invalid or missing in package">Invalid/Missing metadata in package</string>
 
+  <!-- Context: KMP Package strings -->
+  <string name="minimum_keyboard_version_not_supported" comment="Notification when keyboard has functionality not supported by current Keyman">
+    Keyboard package has functionality not supported with this version of Keyman</string>
```
Comment (mcdurdin) [inline android/KMAPro/kMAPro/src/main/res/values/strings.xml:246]:
```suggestion
    Keyboard requires a newer version of Keyman</string>
```

Simpler message may be easier to understand?

**Review by mcdurdin — APPROVED** [review]

**Review by ermshiperete — APPROVED** [review]

**Comment by bharanidharanj** [comment]
* **TEST_FILEVERSION_CHECKED (PASSED):** Tested this as per the instructions (in Android 11 Version- API 30 emulator) and it is working as expected. However, I got different warning message while trying to install test_99.kmp. Attached here for your reference.
